risc0/zkvm/src/claim/receipt.rs

```diff
@@ -346,6 +346,7 @@ impl MaybePruned<WorkClaim<ReceiptClaim>> {
     }
 }
 
+#[cfg_attr(feature = "rv32im-m3", allow(dead_code))]
```

Comment:
Why `allow(dead_code)` instead of conditionally compiling it out?

Reply:
In general I try to avoid compiling things out when possible because I find it to be a better strategy to ensure it continue to compile. Unused code will get deleted by the compiler later on anyway.